pkg/controller/compute/eks.go

```diff
@@ -133,22 +127,23 @@ func (r *Reconciler) fail(instance *awscomputev1alpha2.EKSCluster, err error) (r
 }
 
 func (r *Reconciler) _connect(instance *awscomputev1alpha2.EKSCluster) (eks.Client, error) {
-	// Fetch Provider
 	p := &awsv1alpha2.Provider{}
-	err := r.Get(ctx, meta.NamespacedNameOf(instance.Spec.ProviderReference), p)
-	if err != nil {
+	if err := r.Get(ctx, meta.NamespacedNameOf(instance.Spec.ProviderReference), p); err != nil {
 		return nil, err
 	}
 
-	// Get Provider's AWS Config
-	config, err := awsClient.Config(r.kubeclient, p)
-	if err != nil {
+	s := &v1.Secret{}
+	n := types.NamespacedName{Namespace: p.GetNamespace(), Name: p.Spec.Secret.Name}
+	if err := r.Get(ctx, n, s); err != nil {
 		return nil, err
 	}
 
-	// Connection Region must be with Spec.Region
-	if string(instance.Spec.Region) != config.Region {
-		config.Region = string(instance.Spec.Region)
+	// NOTE(negz): EKS clusters must specify a region for creation. They never
+	// use the provider's region. This should be addressed per the below issue.
+	// https://github.com/crossplaneio/stack-aws/issues/38
+	config, err := aws.LoadConfig(s.Data[p.Spec.Secret.Key], aws.DefaultSection, string(instance.Spec.Region))
```

Comment:
FYI, there is also this method which does exactly that. 
https://github.com/crossplaneio/stack-aws/blob/master/pkg/controller/utils/utils.go#L19

Reply:
There's a slight difference in this one, in that we pass in the instance's region, not the provider's.